lib/ecto_psql_extras/diagnose_logic.ex

```diff
@@ -0,0 +1,222 @@
+    indexes = EctoPSQLExtras.null_indexes(
+      repo,
+      format: :raw,
+      args: [min_relation_size_mb: @null_indexes_min_size_mb]
+    ).rows
+    |> Enum.filter(fn(el) ->
+      null_frac = Enum.at(el, 4)
+      |> String.replace("%", "")
+      |> Float.parse
+      |> elem(0)
+      null_frac > @null_min_null_frac_percent
+    end)
+
+    [ok, message] = case indexes do
+      [] ->
+        [true, "No null indexes detected."]
+      _ ->
+        print_indexes = Enum.map(indexes, fn(el) ->
+          "'#{Enum.at(el, 1)}'"
+        end) |> Enum.join(", ")
+
+        [false, "Null indexes detected: #{print_indexes}"]
+    end
+
+    [ok, "null_indexes", message]
+  end
+
+  defp bloat(repo) do
+    bloated_objects = EctoPSQLExtras.bloat(
+      repo,
+      format: :raw
+    ).rows
+    |> Enum.filter(fn(el) ->
+      Enum.at(el, 3) > Decimal.from_float(@bloat_min_value)
+    end)
+
+    [ok, message] = case bloated_objects do
+      [] ->
+        [true, "No bloated tables or indexes detected."]
+      _ ->
+        print_bloat = Enum.map(bloated_objects, fn(el) ->
+          "#{Enum.at(el, 0)} '#{Enum.at(el, 2)}'"
+        end) |> Enum.join(", ")
+
+        [false, "Bloat detected: #{print_bloat}"]
+    end
+
+    [ok, "bloat", message]
+  end
+
+  defp duplicate_indexes(repo) do
+    indexes = EctoPSQLExtras.duplicate_indexes(
+      repo,
+      format: :raw
+    ).rows
+
+    [ok, message] = case indexes do
+      [] ->
+        [true, "No duplicate indexes detected."]
+      _ ->
+        print_indexes = Enum.map(indexes, fn(el) ->
+          "'#{Enum.at(el, 1)}' - '#{Enum.at(el, 2)}' size #{Enum.at(el, 0)}'"
+        end) |> Enum.join(", ")
+
+        [false, "Duplicate indexes detected: #{print_indexes}"]
+    end
+
+    [ok, "duplicate_indexes", message]
+  end
+
+  defp outliers(repo) do
+    case EctoPSQLExtras.pg_stat_statements_version(repo) do
+      nil ->
+        [false, "outliers", "Cannot check outliers because 'pg_stat_statements' extension is not enabled."]
+      _ ->
+        outliers_data(repo)
+    end
+  end
+
+  defp outliers_data(repo) do
+    queries = EctoPSQLExtras.outliers(
+      repo,
+      format: :raw
+    ).rows
+    |> Enum.filter(fn(el) ->
+      Enum.at(el, 2) > @outliers_min_exec_ratio
+    end)
+
+    [ok, message] = case queries do
+      [] ->
+        [true, "No queries using significant execution ratio detected."]
+      _ ->
+        print_queries = Enum.map(queries, fn(el) ->
+          "'#{Enum.at(el, 0) |> String.slice(0, 20)}...' using #{Enum.at(el, 2)}%"
+        end) |> Enum.join(", ")
+
+        [false, "Queries using significant execution ratio detected: #{print_queries}"]
+    end
+
+    [ok, "outliers", message]
+  end
+
+  defp ssl_used(repo) do
+    case EctoPSQLExtras.ssl_info_enabled(repo) do
+      true ->
+        ssl_used_data(repo)
+      false ->
+        [false, "ssl_used", "Cannot check connection status because 'ssl_info' extension is not enabled."]
+    end
```

Comment:
Just style too:

```suggestion
    if EctoPSQLExtras.ssl_info_enabled(repo) do
      ssl_used_data(repo)
    else
      [false, "ssl_used", "Cannot check connection status because 'ssl_info' extension is not enabled."]
    end
```